fix: update docker-compose to use image built from Dockerfile

justinhynes opened the pull request:
The image pulled and used in devstack by default continues to be the container built from our legacy ansible configuration. This means that developers aren't testing and developing against the same image deployed in Stage and Production.

Additionally, the legacy image is running older versions of Node and npm (Node 16 and npm 8.x, instead of Node 18 and npm 9.9.x).

This PR updates devstack to use the image built from the Credentials dockerfile, which is built after every merge to the `master` branch.

----

I've completed each of the following or determined they are not applicable:

- [ ] Made a plan to communicate any major developer interface changes (or N/A)

jmbowman commented on clone.py:
```diff
@@ -0,0 +1,77 @@
+"""
```
Add a shebang to the start of the script so it can be run standalone? `#!/usr/bin/env python3`

rlucioni replied:
Sure, done.

jmbowman commented on clone.py:
```diff
@@ -0,0 +1,77 @@
+        },
+    },
+})
+logger = logging.getLogger()
```
Any reason for not going with `getLogger(__name__)`?

rlucioni replied:
Nope, just forgot. Added.

jmbowman commented on settings.yml:
```diff
@@ -0,0 +1,5 @@
+repos:
+  - https://github.com/edx/course-discovery.git
```
If I remember correctly, working with HTTPS github URLs with 2FA enabled involves making a personal access token, etc. Any chance of giving the option of doing SSH-style clones instead?

rlucioni replied:
HTTPS clone URLs are currently GitHub's [recommended remote URL](https://help.github.com/articles/which-remote-url-should-i-use/#cloning-with-https-urls-recommended). Cloning with these and 2FA enabled does require a personal access token, but you only need to provide it once, at which point it can be cached by your credential helper of choice (e.g., osxkeychain).

What's the benefit of using SSH URLs at this point? We can add it, but I'd rather not overcomplicate things if we don't need to.

jmbowman replied:
Not a strong requirement, I've just always found SSH checkouts to be easier and more reliable (although that's largely due to experiences with subrepos a few years back)

rlucioni replied:
Okay. I'll leave it out for now, but if it becomes an issue adding it will be straightforward.

justinhynes committed:
fix: update docker-compose to use image built from Dockerfile

The image pulled and used in devstack by default continues to be the container built from our legacy ansible configuration. This means that developers aren't testing and developing against the same image deployed in Stage and Production.

Additionally, the legacy image is running older versions of Node and npm (Node 16 and npm 8.x, instead of Node 18 and npm 9.9.x).

This PR updates devstack to use the image built from the Credentials dockerfile, which is built after every merge to the `master` branch.
